### PR TITLE
Resource: twingate_user

### PR DESCRIPTION
**Pull request by vmanilo**
Resolves https://github.com/Twingate/terraform-provider-twingate/issues/321

## Changes
- added new resource twingate_user
- added unit and acc tests

### CODE REVIEW

File: twingate/internal/provider/resource/user.go [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+			attr.Type: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: fmt.Sprintf("Indicates the User's type. Either %s.", utils.DocList(model.UserTypes)),
+			},
+			attr.ID: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Autogenerated ID of the User, encoded in base64.",
+			},
```
Comment (vmanilo) [inline twingate/internal/provider/resource/user.go:76]: @alexmensch please take a look at descriptions

**Review by alexmensch — APPROVED** [review]
Made copy edits, good to go!

File: twingate/internal/attr/user.go [hunk 2]
```diff
@@ -1,10 +1,12 @@
...
-	FirstName = "first_name"
-	LastName  = "last_name"
-	Email     = "email"
-	IsAdmin   = "is_admin"
-	Role      = "role"
-	Users     = "users"
+	FirstName  = "first_name"
+	LastName   = "last_name"
+	Email      = "email"
+	IsAdmin    = "is_admin"
```
Comment (ekampf) [inline twingate/internal/attr/user.go:7]: Why do we need this if we have `role` ?

Comment (vmanilo) [reply twingate/internal/attr/user.go:7]: not sure, maybe for easy filtering when we use datasource
in the first implementation that we have for user datasource from 16/08/2022 we already have this two fields

Comment (vmanilo) [reply twingate/internal/attr/user.go:7]: found this commit https://github.com/Twingate/terraform-provider-twingate/commit/81807422cac74a92cb61107f83e388bfb8e5b6a6

**Review by ekampf — APPROVED** [review]

**Pull request merged**